### ISSUE TITLE
[BUG]: Add validations to .tag() method

### ISSUE THREAD

**Issue by iyerrama29** [contentstack/datasync-filesystem-sdk#1]
Describe the bug
- **.tag()** method accepts an empty array

To Reproduce

Query a content type 
Create a query that uses to filter by tags
In the query, enter an empty array

Notice: The result is empty. In contentstack-javascript the result is all the entries. Thus, the behaviour is different in both

**Query**

```ts
Stack.contentType('blog')
  .entries()
  .tags([])
  .find()
```

Expected behavior
The query should be invalidated OR throw errors

For future references: If you wish to query for an empty 'Array' field, use **$size** operator inside query
```ts
// example
Stack.contentType('blog')
  .entries()
  .query({tags: {$size: 0}})
```

Regards,
Ramanathan K

**Comment by iyerrama29** [comment]
**A quick update**

- The above issue's patch will be part of hotfix/1.0.2 release.
- The behaviour will be the same as the one received via Contentstack's API response / `contentstack-javascript`

**Comment by iyerrama29** [comment]
Closing this. The has been fixed in hotfix/1.0.2
